addons/token-exchange/dr_enablement.go

```diff
@@ -0,0 +1,46 @@
+package addons
+
+import (
+	"fmt"
+
+	replicationv1alpha1 "github.com/csi-addons/volume-replication-operator/api/v1alpha1"
+	ocsv1 "github.com/openshift/ocs-operator/api/v1"
+	multiclusterv1alpha1 "github.com/red-hat-storage/odf-multicluster-orchestrator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	RookCSIEnableKey              = "CSI_ENABLE_OMAP_GENERATOR"
+	OwnerMirrorPeerKey            = "multicluster.odf.openshift.io/owner-mirrorpeer"
+	MirroringModeKey              = "mirroringMode"
+	SchedulingIntervalKey         = "schedulingInterval"
+	ReplicationSecretNameKey      = "replication.storage.openshift.io/replication-secret-name"
+	ReplicationSecretNamespaceKey = "replication.storage.openshift.io/replication-secret-namespace"
+	Provisioner                   = "openshift-storage.rbd.csi.ceph.com"
+	HostNamePath                  = "/etc/hostname"
+)
+
+func getClient(config *rest.Config) (client.Client, error) {
+	scheme := runtime.NewScheme()
+	if err := corev1.AddToScheme(scheme); err != nil {
+		return nil, fmt.Errorf("failed to add corev1 scheme to runtime scheme: %v", err)
+	}
+	if err := ocsv1.AddToScheme(scheme); err != nil {
+		return nil, fmt.Errorf("failed to add ocsv1 scheme to runtime scheme: %v", err)
+	}
+	if err := replicationv1alpha1.AddToScheme(scheme); err != nil {
+		return nil, fmt.Errorf("failed to add replicationv1alpha1 scheme to runtime scheme: %v", err)
+	}
+	if err := multiclusterv1alpha1.AddToScheme(scheme); err != nil {
+		return nil, fmt.Errorf("failed to add multiclusterv1alpha1 scheme to runtime scheme: %v", err)
+	}
+	cl, err := client.New(config, client.Options{Scheme: scheme})
+	if err != nil {
+		return nil, err
+	}
+
+	return cl, nil
+}
```

Comment:
We don't need this wrapper. We can create clients once during manager initializtion. This add unnecessary management overhead.

Reply:
the green controller uses it so i have moved it to the manager and initializing the client for manager